src/daemon/rpmostreed-utils.cxx

```diff
@@ -609,3 +610,52 @@ rpmostreed_parse_revision (const char  *revision,
 out:
   return ret;
 }
+
+/* Throw an error if there exists systemd inhibitor locks in `block` mode only.
+ * Note: systemd 248 provides a `--check-inhibitors` option, but it also checks
+ * for inhibitors in `delay` mode, which isn't what we want. */
+gboolean
+check_sd_inhibitor_locks (GCancellable    *cancellable,
+                          GError         **error)
+{
+  GDBusConnection *connection = rpmostreed_daemon_connection ();
+  g_autoptr(GVariant) inhibitors_array_tuple = 
+    g_dbus_connection_call_sync (connection, "org.freedesktop.login1", "/org/freedesktop/login1",
+                                 "org.freedesktop.login1.Manager", "ListInhibitors",
+                                 NULL, (const GVariantType*)"(a(ssssuu))",
+                                 G_DBUS_CALL_FLAGS_NONE, -1, cancellable, error);
+  if (!inhibitors_array_tuple)
+    return glnx_prefix_error (error, "Checking systemd inhibitor locks");
+  else if (g_variant_n_children (inhibitors_array_tuple) < 1)
+    return glnx_throw (error, "LoadInhibitors returned empty tuple");
+  g_autoptr(GVariant) inhibitors_array =
+    g_variant_get_child_value (inhibitors_array_tuple, 0);
+
+  char *what;
+  char *who;
+  char *why;
+  char *mode;
+  int num_block_inhibitors = 0;
+  g_autoptr(GString) error_msg = g_string_new(NULL);
+  GVariantIter viter;
+  g_variant_iter_init (&viter, inhibitors_array);
+  while (g_variant_iter_loop (&viter, "(ssssuu)", &what, &who, &why, &mode, NULL, NULL))
+    {
+      // Only consider shutdown inhibitors in `block` mode.
+      if (strstr (what, "shutdown") && strcmp (mode, "block") == 0)
+        if (++num_block_inhibitors == 1)
```

Comment:
Heh prefix increment, don't see those often nowadays. Combined with the previous comment to make it a block, I'd suggest just splitting that out to be easier to read/harder to mess up, e.g.:

```c
{
  if (num_block_inhibitors == 0)
    ...
  num_block_inhibitors++;
}
```

?

Reply:
Haha, yeah I tried pretty hard to not use braces there.
I thought it'd look simpler to not have those two extra lines.